### PR TITLE
Improve action workflow configuration

### PR DESCRIPTION
**Pull request by yixinglu**
- Limit cpu cores for building and testing.
- Use docker action to build image.
- Disable build action for push event.

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -3,12 +3,6 @@ name: docker
 on:
   schedule:
     - cron: '0 18 * * *'
-  push:
-    branches:
-      - 'v[0-9]+.*'
```
Comment (Shylock-Hg) [inline .github/workflows/docker.yaml:8]: Why remove this ?

Comment (dutor) [reply .github/workflows/docker.yaml:8]: Every PR is already required to be tested based on the upstream. At the meantime, it seems that no one cares about the status of the push event, at present.

Of course, we could add it back if necessary in future.

Comment (Shylock-Hg) [reply .github/workflows/docker.yaml:8]: Emmm. Need add it again when need release.

Comment (dutor) [reply .github/workflows/docker.yaml:8]: > Emmm. Need add it again when need release.

This is an awesome idea. We should conceive of a workflow about releasing, including CI, FO and benchmarks, etc.

Comment (dutor) [reply .github/workflows/docker.yaml:8]: I'm gonna mark it down and merge this PR.

Comment (yixinglu) [reply .github/workflows/docker.yaml:8]: this has been covered by release workflow.

**Pull request merged**